memcached.c

```diff
@@ -3440,19 +3467,34 @@ static inline void process_get_command(conn *c, token_t *tokens, size_t ntokens,
                 }
 
                 /* item_get() has incremented it->refcount for us */
-                pthread_mutex_lock(&c->thread->stats.mutex);
-                c->thread->stats.lru_hits[it->slabs_clsid]++;
-                c->thread->stats.get_cmds++;
-                pthread_mutex_unlock(&c->thread->stats.mutex);
+                if (should_touch) {
+                    pthread_mutex_lock(&c->thread->stats.mutex);
+                    c->thread->stats.touch_cmds++;
+                    c->thread->stats.slab_stats[ITEM_clsid(it)].touch_hits++;
+                    pthread_mutex_unlock(&c->thread->stats.mutex);
+                } else {
+                    pthread_mutex_lock(&c->thread->stats.mutex);
+                    c->thread->stats.lru_hits[it->slabs_clsid]++;
+                    c->thread->stats.get_cmds++;
+                    pthread_mutex_unlock(&c->thread->stats.mutex);
+                }
                 *(c->ilist + i) = it;
                 i++;
 
             } else {
-                pthread_mutex_lock(&c->thread->stats.mutex);
-                c->thread->stats.get_misses++;
-                c->thread->stats.get_cmds++;
-                pthread_mutex_unlock(&c->thread->stats.mutex);
-                MEMCACHED_COMMAND_GET(c->sfd, key, nkey, -1, 0);
+                if (should_touch) {
+                    pthread_mutex_lock(&c->thread->stats.mutex);
```

Comment:
same comments as above re: locks, kill the MEMCACHED_COMMAND_TOUCH and move the MEMCACHED_COMMAND_GET back outside of the branch.